### PR TITLE
dnscontrol: fix version, only build the main package

### PR DESCRIPTION
**Pull request by figsoda**
###### Description of changes

before

```
dnscontrol -dev
```

after

```
dnscontrol 3.31.1-dev
```

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/networking/dnscontrol/default.nix [hunk 1]
```diff
@@ -13,6 +13,8 @@ buildGoModule rec {
 
   vendorHash = "sha256-BE/UnJw5elHYmyB+quN89ZkrlMcTjaVN0T2+h8cpPS8=";
 
+  subPackages = [ "." ];
+
```
Comment (Et7f3) [inline pkgs/applications/networking/dnscontrol/default.nix:17]: Can you build all package and move them to separate output ? Or their are not useful ?

**Comment by Et7f3** [comment]
`nix run nixpkgs#nixpkgs-review -- pr 226894` build on x86_64-darwin

File: pkgs/applications/networking/dnscontrol/default.nix (hunk 1 above)
Comment (figsoda) [reply pkgs/applications/networking/dnscontrol/default.nix:17]: before

```console
$ ls result/bin
build  dnscontrol  generate  validate
```

after

```console
$ ls result/bin
dnscontrol
```

I think `dnscontrol` is the only useful package here

Comment (Et7f3) [reply pkgs/applications/networking/dnscontrol/default.nix:17]: They seems to be test https://github.com/StackExchange/dnscontrol/blob/bede608637dcb5de056358e81e28c84f21c9a4ec/documentation/unittests.md#built-in-tests can you at least run it in checkPhase ?

**Review by SuperSandro2000 — COMMENTED** [review]
this can really be one commit where we don't even change 10 lines. Otherwise fine for me.

**Comment by figsoda** [comment]
squashed the commits

**Pull request merged**